### ISSUE TITLE
Benchmarks not running when specifying benchmark name

### ISSUE THREAD

**Issue by TrAyZeN** [bheisler/criterion.rs#776]
When specifying benchmark name with `cargo bench BENCHNAME` benchmarks are not run. It used to work, but I encountered this issue and I can reproduce it on criterion example.

Here is the output of `cargo bench my_benchmark` on criterion example:
```
$ cargo bench my_benchmark
    Finished bench [optimized] target(s) in 0.02s
     Running unittests src/lib.rs (target/release/deps/mycrate-85c68243aaad160d)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running benches/my_benchmark.rs (target/release/deps/my_benchmark-8c0cd1cf45dd6231)
Gnuplot not found, using plotters backend
```

Here are the files to reproduce:

- Cargo.toml
```rs
[package]
name = "mycrate"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dev-dependencies]
criterion = "=0.5.1"

[[bench]]
name = "my_benchmark"
harness = false
```
- src/lib.rs
```rs
#[inline]
pub fn fibonacci(n: u64) -> u64 {
    match n {
        0 => 1,
        1 => 1,
        n => fibonacci(n - 1) + fibonacci(n - 2),
    }
}
```

- benches/my_benchmark.rs
```rs
use criterion::{black_box, criterion_group, criterion_main, Criterion};
use mycrate::fibonacci;

pub fn criterion_benchmark(c: &mut Criterion) {
    c.bench_function("fib 20", |b| b.iter(|| fibonacci(black_box(20))));
}

criterion_group!(benches, criterion_benchmark);
criterion_main!(benches);
```

Rust version: 1.77.2

**Comment by benedikt-schaber** [comment]
I know, that this issue has been open for a long time, just in case somebody stumbles upon this issue in the future:

## Tl;Dr
`criterion.rs` does not filter based on the benchmark name (in this case `my_benchmark`) but based on group, function and argument. In this case only the group is available: `fib 20`. So to filter for it, use something that matches that.

## Details
`criterion.rs` does not filter based on the benchmark name (in this case `my_benchmark`) but based on the `full_id` field of `crate::report::BenchmarkId` (a.k.a. `InternalBenchmarkId`). Which is formed like this:
```Rust
        let full_id = match (&function_id, &value_str) {
            (Some(func), Some(val)) => format!("{}/{}/{}", group_id, func, val),
            (Some(func), &None) => format!("{}/{}", group_id, func),
            (&None, Some(val)) => format!("{}/{}", group_id, val),
            (&None, &None) => group_id.clone(),
        };
```
(From `criterion.rs/src/report.rs`)

If you use `Criterion::bench_function`, without constructing a group yourself, the name becomes the `group_name` (`group_id` above). `function_name` (`func` above) and `parameter` (`val` above) is left empty.

An example using all three would be:
```Rust
let mut group = c.benchmark_group("my_group");
group.bench_with_input(BenchmarkId::new("my_function", "my_argument"), ...)
```

So the `full_id` would here be `my_group/my_function/my_argument`.
